In [46]:
import os 
import cv2
import numpy as np
from keras.models import load_model
from datetime import datetime
import mysql.connector

# Loading the model

In [38]:
model=load_model('second model.h5')

In [39]:
face_detection2=cv2.CascadeClassifier('haarcascade_frontalface_alt2.xml')

# Defining text fonts and such

In [40]:
font                   = cv2.FONT_HERSHEY_SIMPLEX
bottomLeftCornerOfText = (10,500)
fontScale              = 1
fontColor_green        = (0,255,0)
fontColor_red           =(0,0,255)
lineType               = 2

# Updating to database script

In [49]:
def update_database():
    now = datetime.now()
    current_time = now.strftime("%H%M%S")

    today = date.today()
    current_date= today.strftime("%d%m%Y")
    
    
    mydb=mysql.connector.connect(host='localhost',user='root',password='',database='')
    mycursor=mydb.cursor()
    final_query = f"""INSERT INTO attendance(Date,Name,Time) 
                            VALUES 
                            ({current_date},'aliff nabil',{current_time}) """
    mycursor.execute(final_query)
    mydb.commit()
    print('database update')

# Main function for the face recognition application to be run

In [47]:
webcam=cv2.VideoCapture(1)

def face_detector():
    while True:
        success,img=webcam.read()
        
        image=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
        
        face_detector=face_detection2.detectMultiScale(image,scaleFactor=1.1,minNeighbors=3)
        
        for x,y,w,h in face_detector:
            cv2.rectangle(img,(x,y),(x+w,y+h),(0,255,0),1)
            #print('face detected')
            
            face_outline=img[x:x+w,y:y+h]
            face_outline=cv2.resize(face_outline,(83,83))
            face_outline=face_outline/255
            face_outline=np.expand_dims(face_outline,axis=0)
            
            face_prediction=np.argmax(model.predict(face_outline))
            print(face_prediction)
            
        
            if face_prediction==0:
                cv2.putText(img,'aliff nabil',(x,y),font,fontScale,fontColor_red,lineType)
                update_database()
            else:
                cv2.putText(img,'badrul hisham',(x,y),font,fontScale,fontColor_red,lineType)
        
        #show the video
        cv2.imshow('video',img)

        #if press q, the windows will stop and will not break the kernel. then, can edit the script again
        if cv2.waitKey(1)&0xFF==ord('q'):
            cv2.destroyAllWindows() 
            break
          

In [48]:
face_detector()

1
1
0
database update
1
0
database update
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
0
database update
1
1
1
0
database update
0
database update
0
database update
0
database update
0
database update
0
database update
0
database update
0
database update
0
database update
0
database update
0
database update
0
database update


Getting current time and date

In [30]:
from datetime import datetime

now = datetime.now()

current_time = now.strftime("%H%M%S")
print("Current Time =", current_time)

Current Time = 152728


In [29]:
from datetime import date

today = date.today()

# dd/mm/YY
current_date= today.strftime("%d%m%Y")
print("d1 =", current_date)

d1 = 15092021


create table attendance(Date int,Name text,Time int);

insert into attendance(Date,Name,Time) VALUES (15092021,'aliff',151737);